packages/owlbot-bootstrapper/common-container/fetch-api-info.ts

```diff
@@ -0,0 +1,176 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Storage} from '@google-cloud/storage';
+import {ServiceConfigYaml, DriftApi, ReleaseLevel} from './interfaces';
+import yaml from 'js-yaml';
+import {
+  FileNotFoundError,
+  RepositoryFileCache,
+} from '@google-automations/git-file-utils';
+import {logger} from 'gcf-utils';
+export const BUCKET = 'devrel-prod-settings';
+export const DRIFT_APIS_FILE = 'apis.json';
+
+/**
+ * Function that gets api information from a public DRIFT bucket
+ * @param apiId the unique API ID
+ * @param storageClient an instance of Google Cloud Storage
+ * @returns DriftApi info
+ */
+async function getDriftMetadata(
+  apiId: string,
+  storageClient: Storage,
+  bucket: string,
+  file: string
+): Promise<DriftApi> {
+  const contents = await storageClient.bucket(bucket)?.file(file)?.download();
+  let apis;
+  if (contents) {
+    apis = JSON.parse(contents.toString()).apis;
+  }
+
+  return extractApiInfoFromJson(apis, apiId);
+}
+/**
+ * Function that extracts information for a given API from apis.json
+ *
+ * @param apis the apis.json file downloaded from drift as a JSON object
+ * @param apiId the unique identifier of the API
+ * @returns DriftAPI the fields as described in Drift
+ */
+function extractApiInfoFromJson(apis: DriftApi[], apiId: string): DriftApi {
+  const shortName = apiId.split('.')[apiId.split('.').length - 2];
+  for (const api of apis) {
+    if (
+      api.api_shortname === shortName ||
+      api.api_shortname.includes(`cloud${shortName}`)
+    ) {
+      return api;
+    }
+  }
+
+  // If we don't find a match, return an empty object
+  logger.info(
+    `There was no match in apis.json for ${apiId}, shortname: ${shortName}`
+  );
+  return {
+    api_shortname: shortName,
+    display_name: '',
+    docs_root_url: '',
+    launch_stage: 'LAUNCH_STAGE_UNSPECIFIED' as ReleaseLevel,
+    github_label: '',
+  };
+}
+
+/**
+ * Function that gets information from the service_config.yaml file in googleapis/googleapis
+ *
+ * @param apiId unique identifier fo API ID
+ * @param repositoryFileCache wrapper for getting repository contents from github (allows globbing)
+ * @returns a yaml object
+ */
+async function getApiProtoInformation(
+  apiId: string,
+  repositoryFileCache: RepositoryFileCache
+): Promise<Partial<ServiceConfigYaml>> {
+  const path = apiId.toString().replace(/\./g, '/');
+
+  let yamlFile;
+  try {
+    yamlFile = await repositoryFileCache.getFileContents(
+      `${path}/.*?_v.*?.yaml$`,
+      'main'
+    );
```

Comment:
This method is for fetching file contents by an exact file by name. There is a `findFilesByGlob` that can return 0-n matching file names which can be used to discover the file to fetch.